servers/physics_2d/physics_server_2d_sw.cpp

```diff
@@ -1217,10 +1208,8 @@ void PhysicsServer2DSW::set_active(bool p_active) {
 
 void PhysicsServer2DSW::init() {
 	doing_sync = false;
-	last_step = 0.001;
 	iterations = 8; // 8?
 	stepper = memnew(Step2DSW);
-	direct_state = memnew(PhysicsDirectBodyState2DSW);
```

Comment:
I feel it may be a better idea to create this on demand (when requested, if null, create and return)

Reply:
It is used to sync the server state with the Godot scene every iteration; so I think there is little point in delaying its creation:
https://github.com/godotengine/godot/blob/a5fa9199efe0943bc8e852ca300bbc3cd33b25af/servers/physics_2d/body_2d_sw.cpp#L586-L592
BTW This PR simplifies this to simply use it's own copy:
![DirectAccess](https://user-images.githubusercontent.com/9253928/100980049-93dd5d00-353c-11eb-9b30-ce7e534cf50d.png)

Note: `fi_callback` is defined during the `RigidBody` creation:
https://github.com/godotengine/godot/blob/a5fa9199efe0943bc8e852ca300bbc3cd33b25af/scene/2d/physics_body_2d.cpp#L863

Reply:
👍